From 7b76b518036e1438ddc0d3e86ec6ae73e7696e53 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kaj=20Gr=C3=B6nholm?= <kaj.gronholm@qt.io>
Date: Tue, 7 Oct 2025 09:45:51 +0300
Subject: [PATCH 135/239] Assert that overlay isn't null

We expect overlay to always exist at this point.

Pick-to: 6.8
Task-number: QTBUG-139269
Change-Id: I51d986757398be3b4eb246772f37394bc9a443a8
Reviewed-by: Mitch Curtis <mitch.curtis@qt.io>
(cherry picked from commit dd6865c8087bf5aa6fb06ffd62c3348ca8ef0663)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/quicktemplates/qquickpopup.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/quicktemplates/qquickpopup.cpp b/src/quicktemplates/qquickpopup.cpp
index 637d4f8fce..fbcdce4e5a 100644
--- a/src/quicktemplates/qquickpopup.cpp
+++ b/src/quicktemplates/qquickpopup.cpp
@@ -3222,6 +3222,7 @@ bool QQuickPopup::overlayEvent(QQuickItem *item, QEvent *event)
     // If \a item doesn't belong to any of these popups, but exists in an overlay subtree, we shouldn't filter the event,
     // since the item is supposed to be independent of any active popups.
     auto *overlay = QQuickOverlay::overlay(d->window);
+    Q_ASSERT(overlay);
     const QList<QQuickItem *> paintOrderChildItems = QQuickOverlayPrivate::get(overlay)->paintOrderChildItems();
     const qsizetype targetItemPaintOrderIndex = paintOrderChildItems.indexOf(findRootOfOverlaySubtree(item, overlay));
     const qsizetype popupItemPaintOrderIndex = paintOrderChildItems.indexOf(d->popupItem);
-- 
2.51.2

